hooks/useAddTrackToPlaylist.tsx

```diff
@@ -17,18 +24,24 @@ export const useAddTrackToPlaylist = () => {
 
     if (trackExists) {
       console.log("Track already exists in playlist");
-      return;
+      return { success: false, reason: 'TRACK_EXISTS' };
     }
 
-    await sendApiRequest<SpotifyPlaylistItem>({
-      path: `playlists/${todayPlaylistId}/tracks`,
-      method: "POST",
-      body: JSON.stringify({
-        uris: [trackURI],
-      }),
-    });
+    try {
+      await sendApiRequest<SpotifyPlaylistItem>({
+        path: `playlists/${todayPlaylistId}/tracks`,
+        method: "POST",
+        body: JSON.stringify({
+          uris: [trackURI],
+        }),
+      });
 
-    refetchPlaylist();
+      refetchPlaylist();
+      return { success: true };
+    } catch (error) {
+      console.error("Error adding track to playlist:", error);
+      return { success: false, reason: 'API_ERROR' };
```

Comment:
Hmm I'm a bit skeptical with these type of return data in a react hook. The convention when you want to tract status of data fetching is using useState, and return that state instead. For example, you can use states like `isLoading`, `isError` inside these hook, the same goes for error message if you need one. Example as below:
<img width="720" alt="image" src="https://github.com/user-attachments/assets/60928e57-5cab-4e5d-83d2-f482fad96c53" />

For further read: https://www.robinwieruch.de/react-hooks-fetch-data/